fall-2020/navigation.py

```diff
@@ -70,24 +66,117 @@ def set_desired_position(self, desired_position):
         Note: assume the user will pass in the desired_position parameter when using
         the interface 
         """
-        pass
+        self.desired_position = desired_position
+        #we could call the naviagte function here to start movement as soon as desired position given
+        #self.navigate()
+        
 
     
     def update_current_position(self):
         """ Updates the current position of the TBM """
-        pass
+        self.GPS.pollSensor()
+        self.current_position = self.GPS.getPos()
+        
 
 
     def navigate(self):
         """ Navigate to the desired position from the current position
         
         Based on the current position tuple, compared to the desired position tuple,
         make decisions on steering, and ensure that the actuations are successful
-
         Returns: True if actuation requests were successful, False if not
         Note: It may be good to notify the user if something unexpected happens!
         """
-        pass
+        #checking if current postion and desired position are same and if same then prompting to input new values and calling navigate again
+        #if both positions are same then we could either take new input and run navigate again like shown below or we could return an error status like the boolean values for actuatuion requests as mentioned above.
+        if self.current_position == self.desired_position :
+            print("TBM is already at desired position or desired position not set,please set new desired position \n")
+            coordinate = input ("please input desired coordinate in the following form: ' x , y , z ' \n")
+            #coordinate value taken at one go and then seperated as inputting the coordinates seperately will take longer time
+            coordinate.replace(" ","")
+            desired_coordinate_list = coordinate.split(",")
+            desired_position_tuple = ()
+            for number_of_coordinates in range(3):
+                updater_tuple = (desired_coordinate_list[number_of_coordinates],)
+                desired_position_tuple = desired_position_tuple + updater_tuple
+            self.set_desired_position(desired_position_tuple)
+            self.navigate
```

Comment:
Not really sure what this line is supposed to do

Reply:
This is a typo, thanks for pointing it out. After having the new desired position given by the user, I am calling the navigate method again to see if the new position given is acceptable.